platform accton as5710_54x patch

Support for the Accton AS5710_54X networking platform.

diff --git a/arch/powerpc/boot/dts/as5710_54x.dts b/arch/powerpc/boot/dts/as5710_54x.dts
new file mode 100644
index 0000000..0b21a89
--- /dev/null
+++ b/arch/powerpc/boot/dts/as5710_54x.dts
@@ -0,0 +1,1517 @@
+/*
+ * Accton Technology AS5710_54X Device Tree Source
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under  the terms of the GNU General  Public License as published by the
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ *
+ */
+
+/dts-v1/;
+
+/ {
+	compatible = "accton,as5710_54x";
+	#address-cells = <0x2>;
+	#size-cells = <0x2>;
+	interrupt-parent = <0x1>;
+	model = "accton,as5710_54x";
+
+	cpus {
+		power-isa-version = "2.06";
+		power-isa-b;
+		power-isa-e;
+		power-isa-atb;
+		power-isa-cs;
+		power-isa-ds;
+		power-isa-e.ed;
+		power-isa-e.pd;
+		power-isa-e.hv;
+		power-isa-e.le;
+		power-isa-e.pm;
+		power-isa-e.pc;
+		power-isa-ecl;
+		power-isa-exp;
+		power-isa-fp;
+		power-isa-fp.r;
+		power-isa-mmc;
+		power-isa-scpm;
+		power-isa-wt;
+		fsl,eref-deo;
+		mmu-type = "power-embedded";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+
+		PowerPC,e500mc@0 {
+			device_type = "cpu";
+			reg = <0x0>;
+			clocks = <0x2>;
+			next-level-cache = <0x3>;
+			linux,phandle = <0xc>;
+			phandle = <0xc>;
+
+			l2-cache {
+				next-level-cache = <0x4>;
+				linux,phandle = <0x3>;
+				phandle = <0x3>;
+			};
+		};
+
+		PowerPC,e500mc@1 {
+			device_type = "cpu";
+			reg = <0x1>;
+			clocks = <0x5>;
+			next-level-cache = <0x6>;
+			linux,phandle = <0xd>;
+			phandle = <0xd>;
+
+			l2-cache {
+				next-level-cache = <0x4>;
+				linux,phandle = <0x6>;
+				phandle = <0x6>;
+			};
+		};
+
+		PowerPC,e500mc@2 {
+			device_type = "cpu";
+			reg = <0x2>;
+			clocks = <0x7>;
+			next-level-cache = <0x8>;
+			linux,phandle = <0xe>;
+			phandle = <0xe>;
+
+			l2-cache {
+				next-level-cache = <0x4>;
+				linux,phandle = <0x8>;
+				phandle = <0x8>;
+			};
+		};
+
+		PowerPC,e500mc@3 {
+			device_type = "cpu";
+			reg = <0x3>;
+			clocks = <0x9>;
+			next-level-cache = <0xa>;
+			linux,phandle = <0xf>;
+			phandle = <0xf>;
+
+			l2-cache {
+				next-level-cache = <0x4>;
+				linux,phandle = <0xa>;
+				phandle = <0xa>;
+			};
+		};
+	};
+
+	aliases {
+		ccsr = "/soc@ffe000000";
+		dcsr = "/dcsr@f00000000";
+		serial0 = "/soc@ffe000000/serial@11c500";
+		serial1 = "/soc@ffe000000/serial@11c600";
+		serial2 = "/soc@ffe000000/serial@11d500";
+		serial3 = "/soc@ffe000000/serial@11d600";
+		pci0 = "/pcie@ffe200000";
+		pci1 = "/pcie@ffe201000";
+		pci2 = "/pcie@ffe202000";
+		usb0 = "/soc@ffe000000/usb@210000";
+		usb1 = "/soc@ffe000000/usb@211000";
+		dma0 = "/soc@ffe000000/dma@100300";
+		dma1 = "/soc@ffe000000/dma@101300";
+		sdhc = "/soc@ffe000000/sdhc@114000";
+		msi0 = "/soc@ffe000000/msi@41600";
+		msi1 = "/soc@ffe000000/msi@41800";
+		msi2 = "/soc@ffe000000/msi@41a00";
+		crypto = "/soc@ffe000000/crypto@300000";
+		sec_jr0 = "/soc@ffe000000/crypto@300000/jr@1000";
+		sec_jr1 = "/soc@ffe000000/crypto@300000/jr@2000";
+		sec_jr2 = "/soc@ffe000000/crypto@300000/jr@3000";
+		sec_jr3 = "/soc@ffe000000/crypto@300000/jr@4000";
+		rtic_a = "/soc@ffe000000/crypto@300000/rtic@6000/rtic-a@0";
+		rtic_b = "/soc@ffe000000/crypto@300000/rtic@6000/rtic-b@20";
+		rtic_c = "/soc@ffe000000/crypto@300000/rtic@6000/rtic-c@40";
+		rtic_d = "/soc@ffe000000/crypto@300000/rtic@6000/rtic-d@60";
+		sec_mon = "/soc@ffe000000/sec_mon@314000";
+		rman = "/soc@ffe000000/rman@1e0000";
+		pme = "/soc@ffe000000/pme@316000";
+		qman = "/soc@ffe000000/qman@318000";
+		bman = "/soc@ffe000000/bman@31a000";
+		fman0 = "/soc@ffe000000/fman@400000";
+		ethernet0 = "/soc@ffe000000/fman@400000/ethernet@e6000";
+		ethernet1 = "/soc@ffe000000/fman@400000/ethernet@e0000";
+		phy_sgmii_2 = "/soc@ffe000000/fman@400000/mdio@e1120/ethernet-phy@2";
+		phy_sgmii_1f = "/soc@ffe000000/fman@400000/mdio@e1120/ethernet-phy@1f";
+	};
+
+	memory {
+		device_type = "memory";
+	};
+
+	dcsr@f00000000 {
+		ranges = <0x0 0xf 0x0 0x1008000>;
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "fsl,dcsr", "simple-bus";
+
+		dcsr-epu@0 {
+			compatible = "fsl,p2041-dcsr-epu", "fsl,dcsr-epu";
+			interrupts = <0x34 0x2 0x0 0x0 0x54 0x2 0x0 0x0 0x55 0x2 0x0 0x0>;
+			reg = <0x0 0x1000>;
+		};
+
+		dcsr-npc {
+			compatible = "fsl,dcsr-npc";
+			reg = <0x1000 0x1000 0x1000000 0x8000>;
+		};
+
+		dcsr-nxc@2000 {
+			compatible = "fsl,dcsr-nxc";
+			reg = <0x2000 0x1000>;
+		};
+
+		dcsr-corenet {
+			compatible = "fsl,dcsr-corenet";
+			reg = <0x8000 0x1000 0xb0000 0x1000>;
+		};
+
+		dcsr-dpaa@9000 {
+			compatible = "fsl,p2041-dcsr-dpaa", "fsl,dcsr-dpaa";
+			reg = <0x9000 0x1000>;
+		};
+
+		dcsr-ocn@11000 {
+			compatible = "fsl,p2041-dcsr-ocn", "fsl,dcsr-ocn";
+			reg = <0x11000 0x1000>;
+		};
+
+		dcsr-ddr@12000 {
+			compatible = "fsl,dcsr-ddr";
+			dev-handle = <0xb>;
+			reg = <0x12000 0x1000>;
+		};
+
+		dcsr-nal@18000 {
+			compatible = "fsl,p2041-dcsr-nal", "fsl,dcsr-nal";
+			reg = <0x18000 0x1000>;
+		};
+
+		dcsr-rcpm@22000 {
+			compatible = "fsl,p2041-dcsr-rcpm", "fsl,dcsr-rcpm";
+			reg = <0x22000 0x1000>;
+		};
+
+		dcsr-cpu-sb-proxy@40000 {
+			compatible = "fsl,dcsr-e500mc-sb-proxy", "fsl,dcsr-cpu-sb-proxy";
+			cpu-handle = <0xc>;
+			reg = <0x40000 0x1000>;
+		};
+
+		dcsr-cpu-sb-proxy@41000 {
+			compatible = "fsl,dcsr-e500mc-sb-proxy", "fsl,dcsr-cpu-sb-proxy";
+			cpu-handle = <0xd>;
+			reg = <0x41000 0x1000>;
+		};
+
+		dcsr-cpu-sb-proxy@42000 {
+			compatible = "fsl,dcsr-e500mc-sb-proxy", "fsl,dcsr-cpu-sb-proxy";
+			cpu-handle = <0xe>;
+			reg = <0x42000 0x1000>;
+		};
+
+		dcsr-cpu-sb-proxy@43000 {
+			compatible = "fsl,dcsr-e500mc-sb-proxy", "fsl,dcsr-cpu-sb-proxy";
+			cpu-handle = <0xf>;
+			reg = <0x43000 0x1000>;
+		};
+	};
+
+	bman-portals@ff4000000 {
+		ranges = <0x0 0xf 0xf4000000 0x200000>;
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "simple-bus";
+
+		bman-portal@0 {
+			cell-index = <0x0>;
+			compatible = "fsl,bman-portal";
+			reg = <0x0 0x4000 0x100000 0x1000>;
+			interrupts = <0x69 0x2 0x0 0x0>;
+		};
+
+		bman-portal@4000 {
+			cell-index = <0x1>;
+			compatible = "fsl,bman-portal";
+			reg = <0x4000 0x4000 0x101000 0x1000>;
+			interrupts = <0x6b 0x2 0x0 0x0>;
+		};
+
+		bman-portal@8000 {
+			cell-index = <0x2>;
+			compatible = "fsl,bman-portal";
+			reg = <0x8000 0x4000 0x102000 0x1000>;
+			interrupts = <0x6d 0x2 0x0 0x0>;
+		};
+
+		bman-portal@c000 {
+			cell-index = <0x3>;
+			compatible = "fsl,bman-portal";
+			reg = <0xc000 0x4000 0x103000 0x1000>;
+			interrupts = <0x6f 0x2 0x0 0x0>;
+		};
+
+		bman-portal@10000 {
+			cell-index = <0x4>;
+			compatible = "fsl,bman-portal";
+			reg = <0x10000 0x4000 0x104000 0x1000>;
+			interrupts = <0x71 0x2 0x0 0x0>;
+		};
+
+		bman-portal@14000 {
+			cell-index = <0x5>;
+			compatible = "fsl,bman-portal";
+			reg = <0x14000 0x4000 0x105000 0x1000>;
+			interrupts = <0x73 0x2 0x0 0x0>;
+		};
+
+		bman-portal@18000 {
+			cell-index = <0x6>;
+			compatible = "fsl,bman-portal";
+			reg = <0x18000 0x4000 0x106000 0x1000>;
+			interrupts = <0x75 0x2 0x0 0x0>;
+		};
+
+		bman-portal@1c000 {
+			cell-index = <0x7>;
+			compatible = "fsl,bman-portal";
+			reg = <0x1c000 0x4000 0x107000 0x1000>;
+			interrupts = <0x77 0x2 0x0 0x0>;
+		};
+
+		bman-portal@20000 {
+			cell-index = <0x8>;
+			compatible = "fsl,bman-portal";
+			reg = <0x20000 0x4000 0x108000 0x1000>;
+			interrupts = <0x79 0x2 0x0 0x0>;
+		};
+
+		bman-portal@24000 {
+			cell-index = <0x9>;
+			compatible = "fsl,bman-portal";
+			reg = <0x24000 0x4000 0x109000 0x1000>;
+			interrupts = <0x7b 0x2 0x0 0x0>;
+		};
+
+		bman-bpids@0 {
+			compatible = "fsl,bpid-range";
+			fsl,bpid-range = <0x20 0x20>;
+		};
+	};
+
+	qman-portals@ff4200000 {
+		ranges = <0x0 0xf 0xf4200000 0x200000>;
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "simple-bus";
+
+		qman-portal@0 {
+			cell-index = <0x0>;
+			compatible = "fsl,qman-portal";
+			reg = <0x0 0x4000 0x100000 0x1000>;
+			interrupts = <0x68 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x0>;
+		};
+
+		qman-portal@4000 {
+			cell-index = <0x1>;
+			compatible = "fsl,qman-portal";
+			reg = <0x4000 0x4000 0x101000 0x1000>;
+			interrupts = <0x6a 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x1>;
+		};
+
+		qman-portal@8000 {
+			cell-index = <0x2>;
+			compatible = "fsl,qman-portal";
+			reg = <0x8000 0x4000 0x102000 0x1000>;
+			interrupts = <0x6c 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x2>;
+		};
+
+		qman-portal@c000 {
+			cell-index = <0x3>;
+			compatible = "fsl,qman-portal";
+			reg = <0xc000 0x4000 0x103000 0x1000>;
+			interrupts = <0x6e 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x3>;
+		};
+
+		qman-portal@10000 {
+			cell-index = <0x4>;
+			compatible = "fsl,qman-portal";
+			reg = <0x10000 0x4000 0x104000 0x1000>;
+			interrupts = <0x70 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x4>;
+		};
+
+		qman-portal@14000 {
+			cell-index = <0x5>;
+			compatible = "fsl,qman-portal";
+			reg = <0x14000 0x4000 0x105000 0x1000>;
+			interrupts = <0x72 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x5>;
+		};
+
+		qman-portal@18000 {
+			cell-index = <0x6>;
+			compatible = "fsl,qman-portal";
+			reg = <0x18000 0x4000 0x106000 0x1000>;
+			interrupts = <0x74 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x6>;
+		};
+
+		qman-portal@1c000 {
+			cell-index = <0x7>;
+			compatible = "fsl,qman-portal";
+			reg = <0x1c000 0x4000 0x107000 0x1000>;
+			interrupts = <0x76 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x7>;
+		};
+
+		qman-portal@20000 {
+			cell-index = <0x8>;
+			compatible = "fsl,qman-portal";
+			reg = <0x20000 0x4000 0x108000 0x1000>;
+			interrupts = <0x78 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x8>;
+		};
+
+		qman-portal@24000 {
+			cell-index = <0x9>;
+			compatible = "fsl,qman-portal";
+			reg = <0x24000 0x4000 0x109000 0x1000>;
+			interrupts = <0x7a 0x2 0x0 0x0>;
+			fsl,qman-channel-id = <0x9>;
+		};
+
+		qman-fqids@0 {
+			compatible = "fsl,fqid-range";
+			fsl,fqid-range = <0x100 0x100>;
+		};
+
+		qman-fqids@1 {
+			compatible = "fsl,fqid-range";
+			fsl,fqid-range = <0x8000 0x8000>;
+		};
+
+		qman-pools@0 {
+			compatible = "fsl,pool-channel-range";
+			fsl,pool-channel-range = <0x21 0xf>;
+		};
+
+		qman-cgrids@0 {
+			compatible = "fsl,cgrid-range";
+			fsl,cgrid-range = <0x0 0x100>;
+		};
+	};
+
+	soc@ffe000000 {
+		ranges = <0x0 0xf 0xfe000000 0x1000000>;
+		reg = <0xf 0xfe000000 0x0 0x1000>;
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		device_type = "soc";
+		compatible = "simple-bus";
+
+		i2c@118000 {
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			cell-index = <0x0>;
+			compatible = "fsl-i2c";
+			reg = <0x118000 0x100>;
+			interrupts = <0x26 0x2 0x0 0x0>;
+			dfsrr;
+
+			ne1617ads@18 {
+				compatible = "max1617a";
+				reg = <0x18>;
+			};
+
+			eeprom@50 {
+				compatible = "at24,24c256";
+				reg = <0x50>;
+			};
+
+			ds1672@68 {
+				compatible = "ds1672";
+				reg = <0x68>;
+			};
+		};
+
+		i2c@118100 {
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			cell-index = <0x1>;
+			compatible = "fsl-i2c";
+			reg = <0x118100 0x100>;
+			interrupts = <0x26 0x2 0x0 0x0>;
+			dfsrr;
+
+			pca9548@70 {
+				compatible = "pca9548";
+				reg = <0x70>;
+			};
+		};
+
+		i2c@119000 {
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			cell-index = <0x2>;
+			compatible = "fsl-i2c";
+			reg = <0x119000 0x100>;
+			interrupts = <0x27 0x2 0x0 0x0>;
+			dfsrr;
+		};
+
+		i2c@119100 {
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			cell-index = <0x3>;
+			compatible = "fsl-i2c";
+			reg = <0x119100 0x100>;
+			interrupts = <0x27 0x2 0x0 0x0>;
+			dfsrr;
+
+			accton_as5710_54x_cpld@60 {
+				compatible = "acc_as5710_54x_cpld";
+				reg = <0x60>;
+			};
+		};
+
+		usb@211000 {
+			dr_mode = "host";
+			compatible = "fsl-usb2-dr-v1.6", "fsl,mpc85xx-usb2-dr", "fsl-usb2-dr";
+			reg = <0x211000 0x1000>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			interrupts = <0x2d 0x2 0x0 0x0>;
+			fsl,iommu-parent = <0x10>;
+			fsl,liodn-reg = <0x11 0x524>;
+			phy_type = "utmi";
+		};
+
+		fman@400000 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			cell-index = <0x0>;
+			compatible = "fsl,fman", "simple-bus";
+			ranges = <0x0 0x400000 0x100000>;
+			reg = <0x400000 0x100000>;
+			clock-frequency = <0x0>;
+			interrupts = <0x60 0x2 0x0 0x0 0x10 0x2 0x1 0x1>;
+
+			ethernet@e0000 {
+				tbi-handle = <0x12>;
+				phy-handle = <0x13>;
+				phy-connection-type = "sgmii";
+				cell-index = <0x0>;
+				compatible = "fsl,fman-1g-mac";
+				reg = <0xe0000 0x1000>;
+				fsl,port-handles = <0x14 0x15>;
+				ptimer-handle = <0x16>;
+				linux,phandle = <0x28>;
+				phandle = <0x28>;
+				mac-address = [ 00 00 00 00 00 01 ];
+			};
+
+			mdio@e1120 {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				compatible = "fsl,fman-mdio";
+				reg = <0xe1120 0xee0>;
+				interrupts = <0x64 0x1 0x0 0x0>;
+
+				tbi-phy@8 {
+					reg = <0x8>;
+					device_type = "tbi-phy";
+					linux,phandle = <0x12>;
+					phandle = <0x12>;
+				};
+
+				ethernet-phy@2 {
+					reg = <0x2>;
+					linux,phandle = <0x13>;
+					phandle = <0x13>;
+				};
+
+				ethernet-phy@1f {
+					reg = <0x1f>;
+					linux,phandle = <0x18>;
+					phandle = <0x18>;
+				};
+			};
+
+			ethernet@e6000 {
+				tbi-handle = <0x17>;
+				phy-handle = <0x18>;
+				phy-connection-type = "sgmii";
+				cell-index = <0x3>;
+				compatible = "fsl,fman-1g-mac";
+				reg = <0xe6000 0x1000>;
+				fsl,port-handles = <0x19 0x1a>;
+				ptimer-handle = <0x16>;
+				linux,phandle = <0x27>;
+				phandle = <0x27>;
+			};
+
+			mdio@e7120 {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				compatible = "fsl,fman-tbi";
+				reg = <0xe7120 0xee0>;
+				interrupts = <0x64 0x1 0x0 0x0>;
+
+				tbi-phy@8 {
+					reg = <0x8>;
+					device_type = "tbi-phy";
+					linux,phandle = <0x17>;
+					phandle = <0x17>;
+				};
+			};
+
+			cc {
+				compatible = "fsl,fman-cc";
+			};
+
+			muram@0 {
+				compatible = "fsl,fman-muram";
+				reg = <0x0 0x28000>;
+			};
+
+			bmi@80000 {
+				compatible = "fsl,fman-bmi";
+				reg = <0x80000 0x400>;
+			};
+
+			qmi@80400 {
+				compatible = "fsl,fman-qmi";
+				reg = <0x80400 0x400>;
+			};
+
+			port@81000 {
+				cell-index = <0x0>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x81000 0x1000>;
+				fsl,qman-channel-id = <0x46>;
+			};
+
+			port@82000 {
+				cell-index = <0x1>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x82000 0x1000>;
+				fsl,qman-channel-id = <0x47>;
+			};
+
+			port@83000 {
+				cell-index = <0x2>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x83000 0x1000>;
+				fsl,qman-channel-id = <0x48>;
+			};
+
+			port@84000 {
+				cell-index = <0x3>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x84000 0x1000>;
+				fsl,qman-channel-id = <0x49>;
+			};
+
+			port@85000 {
+				cell-index = <0x4>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x85000 0x1000>;
+				status = "disabled";
+				fsl,qman-channel-id = <0x4a>;
+			};
+
+			port@86000 {
+				cell-index = <0x5>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x86000 0x1000>;
+				status = "disabled";
+				fsl,qman-channel-id = <0x4b>;
+			};
+
+			port@87000 {
+				cell-index = <0x6>;
+				compatible = "fsl,fman-port-oh";
+				reg = <0x87000 0x1000>;
+				status = "disabled";
+			};
+
+			policer@c0000 {
+				compatible = "fsl,fman-policer";
+				reg = <0xc0000 0x1000>;
+			};
+
+			keygen@c1000 {
+				compatible = "fsl,fman-keygen";
+				reg = <0xc1000 0x1000>;
+			};
+
+			dma@c2000 {
+				compatible = "fsl,fman-dma";
+				reg = <0xc2000 0x1000>;
+			};
+
+			fpm@c3000 {
+				compatible = "fsl,fman-fpm";
+				reg = <0xc3000 0x1000>;
+			};
+
+			parser@c7000 {
+				compatible = "fsl,fman-parser";
+				reg = <0xc7000 0x1000>;
+			};
+
+			rtc@fe000 {
+				compatible = "fsl,fman-rtc";
+				reg = <0xfe000 0x1000>;
+				linux,phandle = <0x16>;
+				phandle = <0x16>;
+			};
+
+			port@88000 {
+				cell-index = <0x0>;
+				compatible = "fsl,fman-port-1g-rx";
+				reg = <0x88000 0x1000>;
+				linux,phandle = <0x14>;
+				phandle = <0x14>;
+			};
+
+			port@a8000 {
+				cell-index = <0x0>;
+				compatible = "fsl,fman-port-1g-tx";
+				reg = <0xa8000 0x1000>;
+				fsl,qman-channel-id = <0x41>;
+				linux,phandle = <0x15>;
+				phandle = <0x15>;
+			};
+
+			port@89000 {
+				cell-index = <0x1>;
+				compatible = "fsl,fman-port-1g-rx";
+				reg = <0x89000 0x1000>;
+				linux,phandle = <0x1b>;
+				phandle = <0x1b>;
+			};
+
+			port@a9000 {
+				cell-index = <0x1>;
+				compatible = "fsl,fman-port-1g-tx";
+				reg = <0xa9000 0x1000>;
+				fsl,qman-channel-id = <0x42>;
+				linux,phandle = <0x1c>;
+				phandle = <0x1c>;
+			};
+
+			ethernet@e2000 {
+				cell-index = <0x1>;
+				compatible = "fsl,fman-1g-mac";
+				reg = <0xe2000 0x1000>;
+				fsl,port-handles = <0x1b 0x1c>;
+				ptimer-handle = <0x16>;
+			};
+
+			mdio@e3120 {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				compatible = "fsl,fman-tbi";
+				reg = <0xe3120 0xee0>;
+				interrupts = <0x64 0x1 0x0 0x0>;
+			};
+
+			port@8a000 {
+				cell-index = <0x2>;
+				compatible = "fsl,fman-port-1g-rx";
+				reg = <0x8a000 0x1000>;
+				linux,phandle = <0x1d>;
+				phandle = <0x1d>;
+			};
+
+			port@aa000 {
+				cell-index = <0x2>;
+				compatible = "fsl,fman-port-1g-tx";
+				reg = <0xaa000 0x1000>;
+				fsl,qman-channel-id = <0x43>;
+				linux,phandle = <0x1e>;
+				phandle = <0x1e>;
+			};
+
+			ethernet@e4000 {
+				cell-index = <0x2>;
+				compatible = "fsl,fman-1g-mac";
+				reg = <0xe4000 0x1000>;
+				fsl,port-handles = <0x1d 0x1e>;
+				ptimer-handle = <0x16>;
+			};
+
+			mdio@e5120 {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				compatible = "fsl,fman-tbi";
+				reg = <0xe5120 0xee0>;
+				interrupts = <0x64 0x1 0x0 0x0>;
+			};
+
+			port@8b000 {
+				cell-index = <0x3>;
+				compatible = "fsl,fman-port-1g-rx";
+				reg = <0x8b000 0x1000>;
+				linux,phandle = <0x19>;
+				phandle = <0x19>;
+			};
+
+			port@ab000 {
+				cell-index = <0x3>;
+				compatible = "fsl,fman-port-1g-tx";
+				reg = <0xab000 0x1000>;
+				fsl,qman-channel-id = <0x44>;
+				linux,phandle = <0x1a>;
+				phandle = <0x1a>;
+			};
+
+			port@8c000 {
+				cell-index = <0x4>;
+				compatible = "fsl,fman-port-1g-rx";
+				reg = <0x8c000 0x1000>;
+				linux,phandle = <0x1f>;
+				phandle = <0x1f>;
+			};
+
+			port@ac000 {
+				cell-index = <0x4>;
+				compatible = "fsl,fman-port-1g-tx";
+				reg = <0xac000 0x1000>;
+				fsl,qman-channel-id = <0x45>;
+				linux,phandle = <0x20>;
+				phandle = <0x20>;
+			};
+
+			ethernet@e8000 {
+				cell-index = <0x4>;
+				compatible = "fsl,fman-1g-mac";
+				reg = <0xe8000 0x1000>;
+				fsl,port-handles = <0x1f 0x20>;
+				ptimer-handle = <0x16>;
+			};
+
+			mdio@e9120 {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				compatible = "fsl,fman-tbi";
+				reg = <0xe9120 0xee0>;
+				interrupts = <0x64 0x1 0x0 0x0>;
+			};
+
+			port@90000 {
+				cell-index = <0x0>;
+				compatible = "fsl,fman-port-10g-rx";
+				reg = <0x90000 0x1000>;
+				linux,phandle = <0x21>;
+				phandle = <0x21>;
+			};
+
+			port@b0000 {
+				cell-index = <0x0>;
+				compatible = "fsl,fman-port-10g-tx";
+				reg = <0xb0000 0x1000>;
+				fsl,qman-channel-id = <0x40>;
+				linux,phandle = <0x22>;
+				phandle = <0x22>;
+			};
+
+			ethernet@f0000 {
+				cell-index = <0x0>;
+				compatible = "fsl,fman-10g-mac";
+				reg = <0xf0000 0x1000>;
+				fsl,port-handles = <0x21 0x22>;
+			};
+
+			mdio@f1000 {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				compatible = "fsl,fman-xmdio";
+				reg = <0xf1000 0x1000>;
+				interrupts = <0x64 0x1 0x0 0x0>;
+			};
+		};
+
+		soc-sram-error {
+			compatible = "fsl,soc-sram-error";
+			interrupts = <0x10 0x2 0x1 0x1d>;
+		};
+
+		corenet-law@0 {
+			compatible = "fsl,corenet-law";
+			reg = <0x0 0x1000>;
+			fsl,num-laws = <0x20>;
+		};
+
+		memory-controller@8000 {
+			compatible = "fsl,qoriq-memory-controller-v4.5", "fsl,qoriq-memory-controller";
+			reg = <0x8000 0x1000>;
+			interrupts = <0x10 0x2 0x1 0x17>;
+			linux,phandle = <0xb>;
+			phandle = <0xb>;
+		};
+
+		l3-cache-controller@10000 {
+			compatible = "fsl,p2041-l3-cache-controller", "fsl,p4080-l3-cache-controller", "cache";
+			reg = <0x10000 0x1000>;
+			interrupts = <0x10 0x2 0x1 0x1b>;
+			linux,phandle = <0x4>;
+			phandle = <0x4>;
+		};
+
+		corenet-cf@18000 {
+			compatible = "fsl,corenet-cf";
+			reg = <0x18000 0x1000>;
+			interrupts = <0x10 0x2 0x1 0x1f>;
+			fsl,ccf-num-csdids = <0x20>;
+			fsl,ccf-num-snoopids = <0x20>;
+		};
+
+		iommu@20000 {
+			compatible = "fsl,pamu-v1.0", "fsl,pamu";
+			reg = <0x20000 0x4000>;
+			ranges = <0x0 0x20000 0x4000>;
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			interrupts = <0x18 0x2 0x0 0x0 0x10 0x2 0x1 0x1e>;
+
+			pamu@0 {
+				reg = <0x0 0x1000>;
+				fsl,primary-cache-geometry = <0x20 0x1>;
+				fsl,secondary-cache-geometry = <0x80 0x2>;
+				linux,phandle = <0x26>;
+				phandle = <0x26>;
+			};
+
+			pamu@1000 {
+				reg = <0x1000 0x1000>;
+				fsl,primary-cache-geometry = <0x20 0x1>;
+				fsl,secondary-cache-geometry = <0x80 0x2>;
+				linux,phandle = <0x10>;
+				phandle = <0x10>;
+			};
+
+			pamu@2000 {
+				reg = <0x2000 0x1000>;
+				fsl,primary-cache-geometry = <0x20 0x1>;
+				fsl,secondary-cache-geometry = <0x80 0x2>;
+			};
+
+			pamu@3000 {
+				reg = <0x3000 0x1000>;
+				fsl,primary-cache-geometry = <0x20 0x1>;
+				fsl,secondary-cache-geometry = <0x80 0x2>;
+			};
+		};
+
+		pic@40000 {
+			interrupt-controller;
+			#address-cells = <0x0>;
+			#interrupt-cells = <0x4>;
+			reg = <0x40000 0x40000>;
+			compatible = "fsl,mpic", "chrp,open-pic";
+			device_type = "open-pic";
+			clock-frequency = <0x0>;
+			linux,phandle = <0x1>;
+			phandle = <0x1>;
+		};
+
+		timer@41100 {
+			compatible = "fsl,mpic-global-timer";
+			reg = <0x41100 0x100 0x41300 0x4>;
+			interrupts = <0x0 0x0 0x3 0x0 0x1 0x0 0x3 0x0 0x2 0x0 0x3 0x0 0x3 0x0 0x3 0x0>;
+		};
+
+		msi@41600 {
+			compatible = "fsl,mpic-msi";
+			reg = <0x41600 0x200 0x44140 0x4>;
+			msi-available-ranges = <0x0 0x100>;
+			interrupts = <0xe0 0x0 0x0 0x0 0xe1 0x0 0x0 0x0 0xe2 0x0 0x0 0x0 0xe3 0x0 0x0 0x0 0xe4 0x0 0x0 0x0 0xe5 0x0 0x0 0x0 0xe6 0x0 0x0 0x0 0xe7 0x0 0x0 0x0>;
+		};
+
+		msi@41800 {
+			compatible = "fsl,mpic-msi";
+			reg = <0x41800 0x200 0x45140 0x4>;
+			msi-available-ranges = <0x0 0x100>;
+			interrupts = <0xe8 0x0 0x0 0x0 0xe9 0x0 0x0 0x0 0xea 0x0 0x0 0x0 0xeb 0x0 0x0 0x0 0xec 0x0 0x0 0x0 0xed 0x0 0x0 0x0 0xee 0x0 0x0 0x0 0xef 0x0 0x0 0x0>;
+		};
+
+		msi@41a00 {
+			compatible = "fsl,mpic-msi";
+			reg = <0x41a00 0x200 0x46140 0x4>;
+			msi-available-ranges = <0x0 0x100>;
+			interrupts = <0xf0 0x0 0x0 0x0 0xf1 0x0 0x0 0x0 0xf2 0x0 0x0 0x0 0xf3 0x0 0x0 0x0 0xf4 0x0 0x0 0x0 0xf5 0x0 0x0 0x0 0xf6 0x0 0x0 0x0 0xf7 0x0 0x0 0x0>;
+		};
+
+		timer@42100 {
+			compatible = "fsl,mpic-global-timer";
+			reg = <0x42100 0x100 0x42300 0x4>;
+			interrupts = <0x4 0x0 0x3 0x0 0x5 0x0 0x3 0x0 0x6 0x0 0x3 0x0 0x7 0x0 0x3 0x0>;
+		};
+
+		global-utilities@e0000 {
+			compatible = "fsl,qoriq-device-config-1.0";
+			reg = <0xe0000 0xe00>;
+			fsl,has-rstcr;
+			#sleep-cells = <0x1>;
+			fsl,liodn-bits = <0xc>;
+			linux,phandle = <0x11>;
+			phandle = <0x11>;
+		};
+
+		global-utilities@e0e00 {
+			compatible = "fsl,qoriq-pin-control-1.0";
+			reg = <0xe0e00 0x200>;
+			#sleep-cells = <0x2>;
+		};
+
+		global-utilities@e1000 {
+			compatible = "fsl,p2041-clockgen", "fsl,qoriq-clockgen-1.0", "fixed-clock";
+			reg = <0xe1000 0x1000>;
+			clock-frequency = <0x0>;
+			clock-output-names = "sysclk";
+			#clock-cells = <0x0>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			linux,phandle = <0x23>;
+			phandle = <0x23>;
+
+			pll0@800 {
+				#clock-cells = <0x1>;
+				reg = <0x800>;
+				compatible = "fsl,core-pll-clock";
+				clocks = <0x23>;
+				clock-output-names = "pll0", "pll0-div2";
+				linux,phandle = <0x24>;
+				phandle = <0x24>;
+			};
+
+			pll1@820 {
+				#clock-cells = <0x1>;
+				reg = <0x820>;
+				compatible = "fsl,core-pll-clock";
+				clocks = <0x23>;
+				clock-output-names = "pll1", "pll1-div2";
+				linux,phandle = <0x25>;
+				phandle = <0x25>;
+			};
+
+			mux0@0 {
+				#clock-cells = <0x0>;
+				reg = <0x0>;
+				compatible = "fsl,core-mux-clock";
+				clocks = <0x24 0x0 0x24 0x1 0x25 0x0 0x25 0x1>;
+				clock-names = "pll0_0", "pll0_1", "pll1_0", "pll1_1";
+				clock-output-names = "cmux0";
+				linux,phandle = <0x2>;
+				phandle = <0x2>;
+			};
+
+			mux1@20 {
+				#clock-cells = <0x0>;
+				reg = <0x20>;
+				compatible = "fsl,core-mux-clock";
+				clocks = <0x24 0x0 0x24 0x1 0x25 0x0 0x25 0x1>;
+				clock-names = "pll0_0", "pll0_1", "pll1_0", "pll1_1";
+				clock-output-names = "cmux1";
+				linux,phandle = <0x5>;
+				phandle = <0x5>;
+			};
+
+			mux2@40 {
+				#clock-cells = <0x0>;
+				reg = <0x40>;
+				compatible = "fsl,core-mux-clock";
+				clocks = <0x24 0x0 0x24 0x1 0x25 0x0 0x25 0x1>;
+				clock-names = "pll0_0", "pll0_1", "pll1_0", "pll1_1";
+				clock-output-names = "cmux2";
+				linux,phandle = <0x7>;
+				phandle = <0x7>;
+			};
+
+			mux3@60 {
+				#clock-cells = <0x0>;
+				reg = <0x60>;
+				compatible = "fsl,core-mux-clock";
+				clocks = <0x24 0x0 0x24 0x1 0x25 0x0 0x25 0x1>;
+				clock-names = "pll0_0", "pll0_1", "pll1_0", "pll1_1";
+				clock-output-names = "cmux3";
+				linux,phandle = <0x9>;
+				phandle = <0x9>;
+			};
+		};
+
+		global-utilities@e2000 {
+			compatible = "fsl,qoriq-rcpm-1.0";
+			reg = <0xe2000 0x1000>;
+			#sleep-cells = <0x1>;
+		};
+
+		sfp@e8000 {
+			compatible = "fsl,p2041-sfp", "fsl,qoriq-sfp-1.0";
+			reg = <0xe8000 0x1000>;
+		};
+
+		serdes@ea000 {
+			compatible = "fsl,p2041-serdes";
+			reg = <0xea000 0x1000>;
+		};
+
+		dma@100300 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			compatible = "fsl,eloplus-dma";
+			reg = <0x100300 0x4>;
+			ranges = <0x0 0x100100 0x200>;
+			cell-index = <0x0>;
+			fsl,iommu-parent = <0x26>;
+			fsl,liodn-reg = <0x11 0x580>;
+
+			dma-channel@0 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x0 0x80>;
+				cell-index = <0x0>;
+				interrupts = <0x1c 0x2 0x0 0x0>;
+			};
+
+			dma-channel@80 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x80 0x80>;
+				cell-index = <0x1>;
+				interrupts = <0x1d 0x2 0x0 0x0>;
+			};
+
+			dma-channel@100 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x100 0x80>;
+				cell-index = <0x2>;
+				interrupts = <0x1e 0x2 0x0 0x0>;
+			};
+
+			dma-channel@180 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x180 0x80>;
+				cell-index = <0x3>;
+				interrupts = <0x1f 0x2 0x0 0x0>;
+			};
+		};
+
+		dma@101300 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			compatible = "fsl,eloplus-dma";
+			reg = <0x101300 0x4>;
+			ranges = <0x0 0x101100 0x200>;
+			cell-index = <0x1>;
+			fsl,iommu-parent = <0x26>;
+			fsl,liodn-reg = <0x11 0x584>;
+
+			dma-channel@0 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x0 0x80>;
+				cell-index = <0x0>;
+				interrupts = <0x20 0x2 0x0 0x0>;
+			};
+
+			dma-channel@80 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x80 0x80>;
+				cell-index = <0x1>;
+				interrupts = <0x21 0x2 0x0 0x0>;
+			};
+
+			dma-channel@100 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x100 0x80>;
+				cell-index = <0x2>;
+				interrupts = <0x22 0x2 0x0 0x0>;
+			};
+
+			dma-channel@180 {
+				compatible = "fsl,eloplus-dma-channel";
+				reg = <0x180 0x80>;
+				cell-index = <0x3>;
+				interrupts = <0x23 0x2 0x0 0x0>;
+			};
+		};
+
+		sdhc@114000 {
+			compatible = "fsl,p2041-esdhc", "fsl,esdhc";
+			reg = <0x114000 0x1000>;
+			interrupts = <0x30 0x2 0x0 0x0>;
+			clock-frequency = <0x0>;
+			fsl,iommu-parent = <0x10>;
+			fsl,liodn-reg = <0x11 0x530>;
+			sdhci,auto-cmd12;
+		};
+
+		serial@11c500 {
+			cell-index = <0x0>;
+			device_type = "serial";
+			compatible = "fsl,ns16550", "ns16550";
+			reg = <0x11c500 0x100>;
+			clock-frequency = <0x0>;
+			interrupts = <0x24 0x2 0x0 0x0>;
+		};
+
+		serial@11c600 {
+			cell-index = <0x1>;
+			device_type = "serial";
+			compatible = "fsl,ns16550", "ns16550";
+			reg = <0x11c600 0x100>;
+			clock-frequency = <0x0>;
+			interrupts = <0x24 0x2 0x0 0x0>;
+		};
+
+		serial@11d500 {
+			cell-index = <0x2>;
+			device_type = "serial";
+			compatible = "fsl,ns16550", "ns16550";
+			reg = <0x11d500 0x100>;
+			clock-frequency = <0x0>;
+			interrupts = <0x25 0x2 0x0 0x0>;
+		};
+
+		serial@11d600 {
+			cell-index = <0x3>;
+			device_type = "serial";
+			compatible = "fsl,ns16550", "ns16550";
+			reg = <0x11d600 0x100>;
+			clock-frequency = <0x0>;
+			interrupts = <0x25 0x2 0x0 0x0>;
+		};
+
+		gpio@130000 {
+			compatible = "fsl,qoriq-gpio";
+			reg = <0x130000 0x1000>;
+			interrupts = <0x37 0x2 0x0 0x0>;
+			#gpio-cells = <0x2>;
+			gpio-controller;
+		};
+
+		rman@1e0000 {
+			compatible = "fsl,rman";
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			ranges = <0x0 0x1e0000 0x20000>;
+			reg = <0x1e0000 0x20000>;
+			interrupts = <0x10 0x2 0x1 0xb>;
+			fsl,qman-channels-id = <0x62 0x63>;
+
+			inbound-block@0 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x0 0x800>;
+			};
+
+			global-cfg@b00 {
+				compatible = "fsl,rman-global-cfg";
+				reg = <0xb00 0x500>;
+			};
+
+			inbound-block@1000 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x1000 0x800>;
+			};
+
+			inbound-block@2000 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x2000 0x800>;
+			};
+
+			inbound-block@3000 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x3000 0x800>;
+			};
+		};
+
+		usb@210000 {
+			compatible = "fsl-usb2-dr-v1.6", "fsl,mpc85xx-usb2-dr", "fsl-usb2-dr";
+			reg = <0x210000 0x1000>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			interrupts = <0x2c 0x2 0x0 0x0>;
+			phy_type = "utmi";
+			fsl,iommu-parent = <0x10>;
+			fsl,liodn-reg = <0x11 0x520>;
+			port0;
+		};
+
+		crypto@300000 {
+			compatible = "fsl,sec-v4.2", "fsl,sec-v4.0";
+			fsl,sec-era = <0x3>;
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			reg = <0x300000 0x10000>;
+			ranges = <0x0 0x300000 0x10000>;
+			interrupts = <0x5c 0x2 0x0 0x0>;
+			fsl,iommu-parent = <0x10>;
+
+			jr@1000 {
+				compatible = "fsl,sec-v4.2-job-ring", "fsl,sec-v4.0-job-ring";
+				reg = <0x1000 0x1000>;
+				interrupts = <0x58 0x2 0x0 0x0>;
+			};
+
+			jr@2000 {
+				compatible = "fsl,sec-v4.2-job-ring", "fsl,sec-v4.0-job-ring";
+				reg = <0x2000 0x1000>;
+				interrupts = <0x59 0x2 0x0 0x0>;
+			};
+
+			jr@3000 {
+				compatible = "fsl,sec-v4.2-job-ring", "fsl,sec-v4.0-job-ring";
+				reg = <0x3000 0x1000>;
+				interrupts = <0x5a 0x2 0x0 0x0>;
+			};
+
+			jr@4000 {
+				compatible = "fsl,sec-v4.2-job-ring", "fsl,sec-v4.0-job-ring";
+				reg = <0x4000 0x1000>;
+				interrupts = <0x5b 0x2 0x0 0x0>;
+			};
+
+			rtic@6000 {
+				compatible = "fsl,sec-v4.2-rtic", "fsl,sec-v4.0-rtic";
+				#address-cells = <0x1>;
+				#size-cells = <0x1>;
+				reg = <0x6000 0x100>;
+				ranges = <0x0 0x6100 0xe00>;
+
+				rtic-a@0 {
+					compatible = "fsl,sec-v4.2-rtic-memory", "fsl,sec-v4.0-rtic-memory";
+					reg = <0x0 0x20 0x100 0x80>;
+				};
+
+				rtic-b@20 {
+					compatible = "fsl,sec-v4.2-rtic-memory", "fsl,sec-v4.0-rtic-memory";
+					reg = <0x20 0x20 0x200 0x80>;
+				};
+
+				rtic-c@40 {
+					compatible = "fsl,sec-v4.2-rtic-memory", "fsl,sec-v4.0-rtic-memory";
+					reg = <0x40 0x20 0x300 0x80>;
+				};
+
+				rtic-d@60 {
+					compatible = "fsl,sec-v4.2-rtic-memory", "fsl,sec-v4.0-rtic-memory";
+					reg = <0x60 0x20 0x500 0x80>;
+				};
+			};
+		};
+
+		sec_mon@314000 {
+			compatible = "fsl,sec-v4.2-mon", "fsl,sec-v4.0-mon";
+			reg = <0x314000 0x1000>;
+			interrupts = <0x5d 0x2 0x0 0x0>;
+		};
+
+		pme@316000 {
+			compatible = "fsl,pme";
+			reg = <0x316000 0x10000>;
+			interrupts = <0x10 0x2 0x1 0x5>;
+		};
+
+		qman@318000 {
+			compatible = "fsl,qman";
+			reg = <0x318000 0x2000>;
+			interrupts = <0x10 0x2 0x1 0x3>;
+		};
+
+		bman@31a000 {
+			compatible = "fsl,bman";
+			reg = <0x31a000 0x1000>;
+			interrupts = <0x10 0x2 0x1 0x2>;
+		};
+	};
+
+	rapidio@ffe0c0000 {
+		reg = <0xf 0xfe0c0000 0x0 0x11000>;
+		compatible = "fsl,srio";
+		interrupts = <0x10 0x2 0x1 0xb>;
+		#address-cells = <0x2>;
+		#size-cells = <0x2>;
+		fsl,iommu-parent = <0x26>;
+		ranges;
+
+		port1 {
+			ranges = <0x0 0x0 0xc 0x20000000 0x0 0x10000000>;
+			#address-cells = <0x2>;
+			#size-cells = <0x2>;
+			cell-index = <0x1>;
+			fsl,liodn-reg = <0x11 0x510>;
+		};
+
+		port2 {
+			ranges = <0x0 0x0 0xc 0x30000000 0x0 0x10000000>;
+			#address-cells = <0x2>;
+			#size-cells = <0x2>;
+			cell-index = <0x2>;
+			fsl,liodn-reg = <0x11 0x514>;
+		};
+	};
+
+	localbus@ffe124000 {
+		reg = <0xf 0xfe124000 0x0 0x1000>;
+		ranges = <0x0 0x0 0xf 0xe8000000 0x8000000>;
+		compatible = "fsl,p2041-elbc", "fsl,elbc", "simple-bus";
+		interrupts = <0x19 0x2 0x0 0x0>;
+		#address-cells = <0x2>;
+		#size-cells = <0x1>;
+
+		nor@0,0 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			compatible = "cfi-flash";
+			reg = <0x0 0x0 0x8000000>;
+			bank-width = <0x2>;
+			device-width = <0x2>;
+			byteswap;
+
+			partition@0 {
+				reg = <0x0 0x20000>;
+				label = "RCW";
+			};
+
+			partition@1 {
+				reg = <0x20000 0x20000>;
+				label = "hw-info";
+			};
+
+			partition@2 {
+				reg = <0x40000 0xd00000>;
+				label = "onie";
+			};
+
+			partition@3 {
+				reg = <0xd40000 0x1400000>;
+				label = "diag";
+			};
+
+			partition@4 {
+				reg = <0x2140000 0x5e00000>;
+				label = "open";
+			};
+
+			partition@5 {
+				reg = <0x7f40000 0x20000>;
+				label = "Fman-FW";
+			};
+
+			partition@6 {
+				reg = <0x7f60000 0x20000>;
+				label = "uboot-env";
+				env_size = <0x2000>;
+			};
+
+			partition@7 {
+				reg = <0x7f80000 0x80000>;
+				label = "uboot";
+			};
+		};
+	};
+
+	pcie@ffe200000 {
+		reg = <0xf 0xfe200000 0x0 0x1000>;
+		ranges = <0x2000000 0x0 0xd0000000 0x0 0xd0000000 0x0 0x8000000 0x1000000 0x0 0x0 0x0 0xf8000000 0x0 0x10000>;
+		compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
+		device_type = "pci";
+		#size-cells = <0x2>;
+		#address-cells = <0x3>;
+		bus-range = <0x0 0xff>;
+		clock-frequency = <0x1fca055>;
+		interrupts = <0x10 0x2 0x1 0xf>;
+		fsl,iommu-parent = <0x26>;
+		fsl,liodn-reg = <0x11 0x500>;
+
+		pcie@0 {
+			ranges = <0x2000000 0x0 0xe0000000 0x2000000 0x0 0xe0000000 0x0 0x8000000 0x1000000 0x0 0x0 0x1000000 0x0 0x0 0x0 0x10000>;
+			reg = <0x0 0x0 0x0 0x0 0x0>;
+			#interrupt-cells = <0x1>;
+			#size-cells = <0x2>;
+			#address-cells = <0x3>;
+			device_type = "pci";
+			interrupts = <0x10 0x2 0x1 0xf>;
+			interrupt-map-mask = <0xf800 0x0 0x0 0x7>;
+			interrupt-map = <0x0 0x0 0x0 0x1 0x1 0x28 0x1 0x0 0x0 0x0 0x0 0x0 0x2 0x1 0x1 0x1 0x0 0x0 0x0 0x0 0x0 0x3 0x1 0x2 0x1 0x0 0x0 0x0 0x0 0x0 0x4 0x1 0x3 0x1 0x0 0x0>;
+		};
+	};
+
+	pcie@ffe201000 {
+		reg = <0xf 0xfe201000 0x0 0x1000>;
+		ranges = <0x2000000 0x0 0xd8000000 0x0 0xd8000000 0x0 0x8000000 0x1000000 0x0 0x0 0x0 0xf8010000 0x0 0x10000>;
+		compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
+		device_type = "pci";
+		#size-cells = <0x2>;
+		#address-cells = <0x3>;
+		bus-range = <0x0 0xff>;
+		clock-frequency = <0x1fca055>;
+		interrupts = <0x10 0x2 0x1 0xe>;
+		fsl,iommu-parent = <0x26>;
+		fsl,liodn-reg = <0x11 0x504>;
+
+		pcie@0 {
+			ranges = <0x2000000 0x0 0xe0000000 0x2000000 0x0 0xe0000000 0x0 0x8000000 0x1000000 0x0 0x0 0x1000000 0x0 0x0 0x0 0x10000>;
+			reg = <0x0 0x0 0x0 0x0 0x0>;
+			#interrupt-cells = <0x1>;
+			#size-cells = <0x2>;
+			#address-cells = <0x3>;
+			device_type = "pci";
+			interrupts = <0x10 0x2 0x1 0xe>;
+			interrupt-map-mask = <0xf800 0x0 0x0 0x7>;
+			interrupt-map = <0x0 0x0 0x0 0x1 0x1 0x29 0x1 0x0 0x0 0x0 0x0 0x0 0x2 0x1 0x5 0x1 0x0 0x0 0x0 0x0 0x0 0x3 0x1 0x6 0x1 0x0 0x0 0x0 0x0 0x0 0x4 0x1 0x7 0x1 0x0 0x0>;
+		};
+	};
+
+	pcie@ffe202000 {
+		reg = <0xf 0xfe202000 0x0 0x1000>;
+		ranges = <0x2000000 0x0 0xe0000000 0x0 0xe0000000 0x0 0x8000000 0x1000000 0x0 0x0 0x0 0xf8020000 0x0 0x10000>;
+		compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
+		device_type = "pci";
+		#size-cells = <0x2>;
+		#address-cells = <0x3>;
+		bus-range = <0x0 0xff>;
+		clock-frequency = <0x1fca055>;
+		interrupts = <0x10 0x2 0x1 0xd>;
+		fsl,iommu-parent = <0x26>;
+		fsl,liodn-reg = <0x11 0x508>;
+
+		pcie@0 {
+			ranges = <0x2000000 0x0 0xe0000000 0x2000000 0x0 0xe0000000 0x0 0x8000000 0x1000000 0x0 0x0 0x1000000 0x0 0x0 0x0 0x10000>;
+			reg = <0x0 0x0 0x0 0x0 0x0>;
+			#interrupt-cells = <0x1>;
+			#size-cells = <0x2>;
+			#address-cells = <0x3>;
+			device_type = "pci";
+			interrupts = <0x10 0x2 0x1 0xd>;
+			interrupt-map-mask = <0xf800 0x0 0x0 0x7>;
+			interrupt-map = <0x0 0x0 0x0 0x1 0x1 0x2a 0x1 0x0 0x0 0x0 0x0 0x0 0x2 0x1 0x9 0x1 0x0 0x0 0x0 0x0 0x0 0x3 0x1 0xa 0x1 0x0 0x0 0x0 0x0 0x0 0x4 0x1 0xb 0x1 0x0 0x0>;
+		};
+	};
+
+	fsl,dpaa {
+		compatible = "fsl,p2041-dpaa", "fsl,dpaa";
+
+		ethernet@0 {
+			compatible = "fsl,p2041-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <0x27>;
+		};
+
+		ethernet@1 {
+			compatible = "fsl,p2041-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <0x28>;
+		};
+	};
+};
diff --git a/arch/powerpc/platforms/85xx/Kconfig b/arch/powerpc/platforms/85xx/Kconfig
index 48a56fd..4deca00 100644
--- a/arch/powerpc/platforms/85xx/Kconfig
+++ b/arch/powerpc/platforms/85xx/Kconfig
@@ -232,6 +232,20 @@ config GE_IMP3A
 	  This board is a 3U CompactPCI Single Board Computer with a Freescale
 	  P2020 processor.
 
+config AS5710_54X
+	bool "Accton Technology Corporation AS5710_54X"
+	select DEFAULT_UIMAGE
+	select PPC_E500MC
+	select PHYS_64BIT
+	select SWIOTLB
+	select ARCH_REQUIRE_GPIOLIB
+	select GPIO_MPC8XXX
+	select HAS_RAPIDIO
+	select PPC_EPAPR_HV_PIC
+	select HAS_FSL_QBMAN
+	help
+	  This option enables support for the ACCTON AS5710_54X networking platform
+
 config P2041_RDB
 	bool "Freescale P2041 RDB"
 	select DEFAULT_UIMAGE
diff --git a/arch/powerpc/platforms/85xx/Makefile b/arch/powerpc/platforms/85xx/Makefile
index e018200..780c7f7 100644
--- a/arch/powerpc/platforms/85xx/Makefile
+++ b/arch/powerpc/platforms/85xx/Makefile
@@ -24,6 +24,7 @@ obj-$(CONFIG_P1022_DS)    += p1022_ds.o
 obj-$(CONFIG_P1022_RDK)   += p1022_rdk.o
 obj-$(CONFIG_P1023_RDS)   += p1023_rds.o
 obj-$(CONFIG_TWR_P102x)   += twr_p102x.o
+obj-$(CONFIG_AS5710_54X)  += as5710_54x.o corenet_ds.o
 obj-$(CONFIG_P2041_RDB)   += p2041_rdb.o corenet_ds.o
 obj-$(CONFIG_P3041_DS)    += p3041_ds.o corenet_ds.o
 obj-$(CONFIG_P4080_DS)    += p4080_ds.o corenet_ds.o
diff --git a/arch/powerpc/platforms/85xx/as5710_54x.c b/arch/powerpc/platforms/85xx/as5710_54x.c
new file mode 100644
index 0000000..6220b1e
--- /dev/null
+++ b/arch/powerpc/platforms/85xx/as5710_54x.c
@@ -0,0 +1,88 @@
+/*
+ * Accton as5710_54x Setup
+ *
+ * Copyright 2011 Freescale Semiconductor Inc.
+ *
+ * This program is free software; you can redistribute	it and/or modify it
+ * under  the terms of	the GNU General	 Public License as published by the
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ */
+
+#include <linux/kernel.h>
+#include <linux/pci.h>
+#include <linux/kdev_t.h>
+#include <linux/delay.h>
+#include <linux/interrupt.h>
+#include <linux/phy.h>
+
+#include <asm/time.h>
+#include <asm/machdep.h>
+#include <asm/pci-bridge.h>
+#include <mm/mmu_decl.h>
+#include <asm/prom.h>
+#include <asm/udbg.h>
+#include <asm/mpic.h>
+
+#include <linux/of_platform.h>
+#include <sysdev/fsl_soc.h>
+#include <sysdev/fsl_pci.h>
+#include <asm/ehv_pic.h>
+
+#include "corenet_ds.h"
+
+/*
+ * Called very early, device-tree isn't unflattened
+ */
+static int __init as5710_54x_probe(void)
+{
+	unsigned long root = of_get_flat_dt_root();
+#ifdef CONFIG_SMP
+	extern struct smp_ops_t smp_85xx_ops;
+#endif
+
+	if (of_flat_dt_is_compatible(root, "accton,as5710_54x"))
+		return 1;
+
+	/* Check if we're running under the Freescale hypervisor */
+	if (of_flat_dt_is_compatible(root, "accton,as5710_54x-hv")) {
+		ppc_md.init_IRQ = ehv_pic_init;
+		ppc_md.get_irq = ehv_pic_get_irq;
+		ppc_md.restart = fsl_hv_restart;
+		ppc_md.power_off = fsl_hv_halt;
+		ppc_md.halt = fsl_hv_halt;
+#ifdef CONFIG_SMP
+		/*
+		 * Disable the timebase sync operations because we can't write
+		 * to the timebase registers under the hypervisor.
+		  */
+		smp_85xx_ops.give_timebase = NULL;
+		smp_85xx_ops.take_timebase = NULL;
+#endif
+		return 1;
+	}
+
+	return 0;
+}
+
+define_machine(as5710_54x) {
+	.name			= "Accton Technology Corporation AS5710_54X",
+	.probe			= as5710_54x_probe,
+	.setup_arch		= corenet_ds_setup_arch,
+	.init_IRQ		= corenet_ds_pic_init,
+#ifdef CONFIG_PCI
+	.pcibios_fixup_bus	= fsl_pcibios_fixup_bus,
+#endif
+	.get_irq		= mpic_get_coreint_irq,
+	.restart		= fsl_rstcr_restart,
+	.calibrate_decr		= generic_calibrate_decr,
+	.progress		= udbg_progress,
+	.power_save		= e500_idle,
+	.init_early		= corenet_ds_init_early,
+};
+
+machine_arch_initcall(as5710_54x, corenet_ds_publish_devices);
+
+#ifdef CONFIG_SWIOTLB
+machine_arch_initcall(as5710_54x, swiotlb_setup_bus_notifier);
+#endif
